File: Front End/script/models/Carro.js
export default class Carro {
  constructor(props = {}) {
    Object.assign(this, {
      marca: "",
      modelo: "",
      ano: 0,
      km: 0,
      combustivel: "",
      transmissao: "",
      motor: "",
      potencia: 0,
      torque: "",
      consumo: "",
      tipo: "",
      condicao: "",
      imagem: "",
      preco: 0,
      ...props
    })
  }
}

File: Back End/src/main/java/com/example/automarket/domain/Carro.java
package com.example.automarket.domain;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;

@Entity
@Table(name = "carros")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(of = "id")
public class Carro {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String marca;

    private String modelo;

    private String tipo;

    private Integer ano;

    private BigDecimal preco;

    @Enumerated(EnumType.STRING)
    private Condicao condicao;

    private String imagem;

    @Embedded
    private FichaTecnica fichaTecnica;

}
